velox/functions/prestosql/JsonFunctions.cpp

```diff
@@ -641,10 +700,52 @@ class JsonExtractFunction : public exec::VectorFunction {
     resultWriter.finish();
   }
 
-  FOLLY_ALWAYS_INLINE simdjson::error_code processJsonExtract(
+  FOLLY_ALWAYS_INLINE static simdjson::error_code processJsonArrayGet(
+      const StringView& jsonArray,
+      int64_t index,
+      std::string& output) {
+    simdjson::ondemand::document jsonDoc;
+
+    simdjson::padded_string paddedJson(jsonArray.data(), jsonArray.size());
+    if (simdjsonParse(paddedJson).get(jsonDoc)) {
+      return simdjson::TAPE_ERROR;
+    }
+    if (jsonDoc.type().error()) {
+      return simdjson::TAPE_ERROR;
+    }
+
+    if (jsonDoc.type() != simdjson::ondemand::json_type::array) {
+      return simdjson::TAPE_ERROR;
+    }
+
+    size_t numElements;
+    if (jsonDoc.count_elements().get(numElements)) {
```

Comment:
nit: why not use SIMDJSON_ASSIGN_OR_RAISE

Reply:
that macro expects a function that returns simdjson_result whereas here it returns just the error code.